creusot-contracts/src/std/slice.rs

```diff
@@ -274,6 +277,9 @@ extern_spec! {
             forall<j:usize> i <= j && @j < (@self).len() ==> x.deep_model() < self.deep_model()[@j])]
         fn binary_search(&self, x : &T) -> Result<usize, usize>
             where T: Ord + DeepModel,  T::DeepModelTy: OrdLogic,;
+
+        #[ensures(@result == @self_)]
+        fn into_vec<A: Allocator>(self_: Box<Self, A>) -> Vec<T, A>;
```

Comment:
I think when we update to `syn 2.0` we'll be able to fix this glitch in the macros that forces us to write `self_`. previous versions of `syn` did not allow custom receiver types.